### PR TITLE
Add contest search functionality

### PR DESCRIPTION
**Pull request by miguelrodrigues22**

### CODE REVIEW

File: pages/index.tsx [hunk 1]
```diff
@@ -8,35 +8,46 @@ import ContestCard from '../components/ContestCard'
 export default function Home() {
   const { loading, error, data } = useQuery(GET_CONTEST_LIST)
 
-  const [statusFilter, setStatusFilter] = useState<IFilter>('open')
+  console.log('data', data)
```
Comment (FilipaSdS) [inline pages/index.tsx:11]: remove console.log please

File: pages/index.tsx [hunk 2]
```diff
@@ -8,35 +8,46 @@ import ContestCard from '../components/ContestCard'
...
+  let filteredContestList = []
+
+  if (!isSearching) {
+    filteredContestList =
+      statusFilter === 'All'
+        ? contestListing
+        : contestListing?.filter((contest) => contest.status === statusFilter)
+  } else if (isSearching) {
+    filteredContestList = searchData
+  }
```
Comment (FilipaSdS) [inline pages/index.tsx:29]: Shouldn't we be able to filter the searched results? (refine the search)

File: pages/index.tsx [hunk 3]
```diff
@@ -8,35 +8,46 @@ import ContestCard from '../components/ContestCard'
...
             statusFilter={statusFilter}
             setStatusFilter={setStatusFilter}
+            setIsSearching={setIsSearching}
           />
         </div>
         <main className='min-h-screen py-8 px-20 flex-1 flex flex-col'>
-          {loading && <p>Loading</p>}
-          {error && <p>Error while retrieving the contests</p>}
+          {/* {loading && <p>Loading</p>}
+          {error && <p>Error while retrieving the contests</p>} */}
```
Comment (FilipaSdS) [inline pages/index.tsx:46]: why is this commented out?

**Review by FilipaSdS — APPROVED** [review]

File: components/helpers/interfaces.tsx [hunk 4]
```diff
@@ -59,4 +59,6 @@ export interface ContestFilterProps {
   setOpen: React.Dispatch<React.SetStateAction<boolean>>
   statusFilter: IFilter
   setStatusFilter: React.Dispatch<React.SetStateAction<IFilter>>
+  setSearchData?: React.Dispatch<React.SetStateAction<any>>
+  setIsSearching?: React.Dispatch<React.SetStateAction<boolean>>
```
Comment (JdFSilva) [inline components/helpers/interfaces.tsx:63]: props should be defined on the same file as the component

**Review by JdFSilva — CHANGES_REQUESTED** [review]

File: components/helpers/interfaces.tsx (hunk 4 above)
Comment (miguelrodrigues22) [reply components/helpers/interfaces.tsx:63]: fixed

**Review by JdFSilva — APPROVED** [review]

**Pull request merged**